src/PluginForm/AbstractPayment.php

```diff
@@ -0,0 +1,207 @@
+      case 'afterpay':
+        $form['birthdate'] = [
+          '#type' => 'date',
+          '#title' => $this->t('Please, set your birthday date'),
+          '#required' => true
+        ];
+        $form['gender'] = [
+          '#type' => 'select',
+          '#title' => $this->t('Choose gender'),
+          '#options' => [$this->t('Male'), $this->t('Female')],
+          '#value' => ['male', 'female']
+        ];
+        if ($this->paymentMethod == 'afterpay') {
+          $link = $this->bankconfig->getAfterPayTermsLink($this->buildersRedefiner->getLangCode($payment));
+          $form['verified_terms_of_service'] = [
+            '#type' => 'checkbox',
+            '#title' => $this->t('I accept '). '<a href="'.$link.'" target="_blank">'.$this->t('Terms and Conditions').'</a>',
+            '#required' => true
+          ];
+        }
+        $form = $this->helper->setDefaultButtons($form);
+        break;
+      case 'bank-transfer':
+        $order = $this->startTransaction($form, $form_state);
+        $form = $this->helper->setBanktransferForm($form, $order->toArray());
+        break;
+      default:
+        $this->startTransaction($form, $form_state);
+        break;
+    }
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateConfigurationForm(array &$form, FormStateInterface $form_state)
+  {
+
+    if ($this->entity->getPaymentGateway()->getPluginId() == 'bank-transfer') {
+      $order = $this->entity->getOrder();
+      $params['subject'] = 'Order payment information!';
+      $params['body'] = $form['banktransfer_info'];
+      $options = array(
+        'langcode' => 'en',
+      );
+      $message['from'] = \Drupal::config('system.site')->get('mail');
+      \Drupal::service(
+        'plugin.manager.mail')->mail('commerce_ginger',
+        'order_mail',
+        $order->getEmail(),
+        $this->buildersRedefiner->getLangCode($this->entity),
+        $params
+      );
+    }
+    $this->startTransaction($form, $form_state);
+  }
+
+
+  public function getIssuerId($values, FormStateInterface $form_state)
+  {
+    $issuer = $values['issuers'];
+    return $issuer[$form_state->getUserInput()["payment_process"]["offsite_payment"]['issuers']];
+  }
+
+  public function getCustomerGender($values, FormStateInterface $form_state)
+  {
+    $issuer = $values['gender'];
+    return $issuer[$form_state->getUserInput()["payment_process"]["offsite_payment"]['gender']];
+  }
+
+  public function getBirthdate($values)
+  {
+    return $values['birthdate'];
+  }
+
+  public function getTermsState($values)
+  {
+    return $values['verified_terms_of_service'] == 1 ? true : false;
+  }
+
+  public function startTransaction(array &$form, FormStateInterface $form_state)
+  {
+    $this->paymentMethod = $this->entity->getPaymentGateway()->getPluginId();
+    $payment = $this->entity;
+    $issuerId = null;
+    $birthdate = null;
+    $gender = null;
+    $verifiedTerms = null;
```

Comment:
Do we really need this inits? Looks like createOrder func expects that those vars could be null, I think the code could work without them (but it needs to be tested)